DQM/CTPPS/plugins/CTPPSDiamondDQMSource.cc

```diff
@@ -0,0 +1,854 @@
+
+      // Check Event Number
+      for ( const auto& optorx : *fedInfo ) {
+        if ( detId.arm() == 1 && optorx.getFEDId() == 582 ) {
+          potPlots_[detId_pot].ECCheck->Fill((int)((optorx.getLV1()& 0xFF)-((unsigned int) status.getEC() & 0xFF)) & 0xFF);
+          if ( ( static_cast<int>( ( optorx.getLV1() & 0xFF )-status.getEC() ) != EC_difference_582_ ) && ( static_cast<uint8_t>( ( optorx.getLV1() & 0xFF )-status.getEC() ) < 128 ) )
+            EC_difference_582_ = static_cast<int>( optorx.getLV1() & 0xFF )-( static_cast<unsigned int>( status.getEC() ) & 0xFF );
+          if ( EC_difference_582_ != 1 && EC_difference_582_ != -500 && EC_difference_582_ < 128 && EC_difference_582_ > -128 )
+            if (verbosity_) edm::LogProblem("CTPPSDiamondDQMSource")  << "FED 852: ECError at EV: 0x"<< std::hex << optorx.getLV1()
+                                                                      << "\t\tVFAT EC: 0x"<< static_cast<unsigned int>( status.getEC() )
+                                                                      << "\twith ID: " << std::dec << detId
+                                                                      << "\tdiff: " <<  EC_difference_582_;
+        }
+        else if ( detId.arm() == 0 && optorx.getFEDId()==583 ) {
+          potPlots_[detId_pot].ECCheck->Fill((int)((optorx.getLV1()& 0xFF)-status.getEC()) & 0xFF);
+          if ( ( static_cast<int>( ( optorx.getLV1() & 0xFF )-status.getEC() ) != EC_difference_583_ ) && ( static_cast<uint8_t>( ( optorx.getLV1() & 0xFF )-status.getEC() ) < 128 ) )
+            EC_difference_583_ = static_cast<int>( optorx.getLV1() & 0xFF )-( static_cast<unsigned int>( status.getEC() ) & 0xFF );
+          if ( EC_difference_583_ != 1 && EC_difference_583_ != -500 && EC_difference_583_ < 128 && EC_difference_583_ > -128 )
+            if (verbosity_) edm::LogProblem("CTPPSDiamondDQMSource")  << "FED 853: ECError at EV: 0x"<< std::hex << optorx.getLV1()
+                                                                      << "\t\tVFAT EC: 0x"<< static_cast<unsigned int>( status.getEC() )
+                                                                      << "\twith ID: " << std::dec << detId
+                                                                      << "\tdiff: " <<  EC_difference_583_;
+        }
+      }
+    }
+  }
+
+  // Using CTPPSDiamondRecHit
+  std::map<unsigned int, std::set<unsigned int> > planes;
+
+  for ( const auto& rechits : *diamondRecHits ) {
+    CTPPSDiamondDetId detId_pot( rechits.detId() );
+    detId_pot.setPlane( 0 );
+    detId_pot.setChannel( 0 );
+    const CTPPSDiamondDetId detId( rechits.detId() );
+
+    for ( const auto& rechit : rechits ) {
+      if ( excludeMultipleHits_ && rechit.getMultipleHits() > 0 ) continue;
+      planes[detId_pot].insert( detId.plane() );
+
+      if ( potPlots_.find( detId_pot ) == potPlots_.end() ) continue;
+
+      const double rnd = ( 2. * randomGen.Rndm() ) - 1.;
```

Comment:
That is true, maybe we can fix the seed at the beginning of the run or lumisection?
We have also the plots with the pad number, the problem is that they are not ordered (due to the channel arrangement on the detector) and of different size: they give good information about our front-end but they are difficult to visually reinterpret to understand the hit distribution.
The trouble is due to the fact that we have relatively large pads and of different size: unusual situation!